docs/content/guide/i18n.ngdoc

```diff
@@ -142,84 +144,179 @@ displaying the date with a timezone specified by the developer.
-examples will make this clear.  In the following example, I
-will only be showing you the AngularJS syntax.
+#### Messages
 
+Messages immediately follow a selection keyword and are optionally preceded by whitespace.  They are
+written in single curly braces (`{}`).  They may contain Angular interpolation syntax inside them.
+In addition, the `#` symbol is a placeholder for the actual numeric value of the expression.
 
 ### Simple plural example
 
-```
+```text
 {{numMessages, plural,
-      =0 { You have no new messages }
-      =1 { You have one new message }
-   other { You have # new messages }
+      =0 {You have no new messages}
+      =1 {You have one new message}
+   other {You have # new messages}
 }}
 ```
 
-While I won't be teaching you MessageFormat here, you will
-note that the `#` symbol works as expected.  You could have
-also written it as:
+Because these messages can themselves contain Angular expressions, you could also write this as
+follows:
 
-```
+```text
 {{numMessages, plural,
-      =0 { You have no new messages }
-      =1 { You have one new message }
-   other { You have {{numMessages}} new messages }
+      =0 {You have no new messages}
+      =1 {You have one new message}
+   other {You have {{numMessages}} new messages}
 }}
 ```
 
-where you explicitly typed in `numMessages` for "other"
-instead of using `#`.  They are nearly the same except if
-you're using "offset".  Refer to the ICU MessageFormat
-documentation to learn about offset.
 
-Please note that **other** is a **required** category (for
-both the plural syntax and the select syntax that is shown
-later.)
+### Plural syntax with optional `offset`
+
+The plural syntax supports an optional `offset` syntax that is used in matching.  It's simpler to
+explain this with an example.
+
+```text
+{{recipients.length, plural, offset:1
+    =0    {You gave no gifts}
+    =1    {You gave {{recipients[0].name}} a gift}
+    one   {You gave {{recipients[0].name}} and one other person a gift}
+    other {You gave {{recipients[0].name}} and # other people a gift}
+}}
+```
+
+When an `offset` is specified, the matching works as follows.  First, the exact value of the Angular
+expression is matched against the exact matches (i.e. `=N` selectors) to find a match.  If there is
+one, that message is used.  If there was no match, then the offset value is subtracted from the
+value of the expression and locale specific pluralization rules are applied to this new value to
+obtain its plural category (such as “one”, “few”, “many”, etc.) and a match is attempted against the
+keyword selectors and the matching message is used.  If there was no match, then the “other”
+category (required) is used.  The value of the `#` character inside a message is the value of
+original expression reduced by the offset value that was specified.
+
+### Escaping / Quoting
+
+You might need to quote the curly braces or the `#` character inside the message texts so that they
```

Comment:
Erh...I think this comment is not valid. (Unless you mean the whole sentence should be changed/re-phrased.)